lib/ciphersuite.ml

```diff
@@ -37,22 +38,31 @@ type aead_cipher =
   | AES_256_GCM
   [@@deriving sexp]
 
-type payload_protection =
-  | Stream of stream_cipher * Nocrypto.Hash.hash
-  | Block of block_cipher * Nocrypto.Hash.hash
-  | AEAD of aead_cipher
-  [@@deriving sexp]
+(* this is K_LEN, max 8 N_MIN from RFC5116 -- as defined in TLS1.3 spec *)
+let kn = function
+  | AES_128_GCM -> (16, 12)
+  | AES_256_GCM -> (32, 12)
+  | AES_128_CCM -> (16, 12)
+  | AES_256_CCM -> (32, 12)
+
+type payload_protection13 = [ `AEAD of aead_cipher ] [@@deriving sexp]
+
+type payload_protection = [
+  payload_protection13
+  | `Stream of stream_cipher * Nocrypto.Hash.hash
+  | `Block of block_cipher * Nocrypto.Hash.hash
+  ] [@@deriving sexp]
 
 (** [key_length iv payload_protection] is [(key size, IV size, mac size)] where key IV, and mac sizes are the required bytes for the given [payload_protection] *)
 let key_length iv pp =
```

Comment:
As discussed it would be nice to have a comment here explaining that this is only used for TLS `<= 1.2 `

Reply:
NB the IV for AEAD stems from https://tools.ietf.org/html/rfc5288#section-3 (for GCM) -- `salt[4]` (CCM https://tools.ietf.org/html/rfc6655#section-3)